### PR TITLE
v2.0.0: Implemented and moved all examples, documentation and tests over to namespaces data access attributes

### PR DESCRIPTION
**Pull request by AndriyMulyar**
# v2.0.0

Atlas map state is now accessible as top level attributes of the AtlasProjection ("the Map") class.
```python
from atlas import AtlasProject

project = AtlasProject(name='My Project')
map = project.maps[0]

```

- Topics (`map.topics`)
- Embeddings (`map.embeddings`)
- Semantic Duplicate Clusters (`map.duplicates`)

You can learn how to use the operations these attributes encapsulate on their corresponding data in the `Accessing Atlas State` section of the documentation.

------
Updated documentation and example scripts to reflect this change.

### CODE REVIEW

File: docs/map_state/embeddings.md
Comment (zanussbaum) [inline docs/map_state/embeddings.md:14]: My first natural question here is what columns can we expect? Will we get both the ambient and 2d representations in the same dataframe?

File: docs/map_state/tags.md
Comment (zanussbaum) [inline docs/map_state/tags.md:16]: Should we note here that tags returns a dictionary where the keys are the tag names the user adds in the frontend and the values are a list of ids? I found this confusing when I was initially using `get_tags`.

Should we also mention that if the use doesn't provide `id_field` in `map_text`, it's very tough to map from tags to your original dataframe? cc @bmschmidt

File: docs/map_state/topics.md
Comment (zanussbaum) [inline docs/map_state/topics.md:15]: nit: same thing here, can we either show an example or explain what the dataframe looks like (e.g. what columns)

File: nomic/atlas.py
Comment (zanussbaum) [inline nomic/atlas.py:81]: it would be great if we either enforce users to upload with an id field or find a way to pass that to the user so they can map back from `datum_id` to their original data. I've had to rebuild maps a few too many times since I forgot to add an id column and couldn't join with the tags I had added to the data points

File: nomic/data_operations.py
Comment (zanussbaum) [inline nomic/data_operations.py:321]: Should we update the documentation if this isn't finished yet?

File: nomic/data_operations.py
Comment (zanussbaum) [inline nomic/data_operations.py:602]: do we want a `__repr__` method to stay consistent with the other classes?

File: nomic/data_operations.py
Comment (zanussbaum) [inline nomic/data_operations.py:433]: nit: is this complete?

File: nomic/gpt4all/gpt4all.py
Comment (zanussbaum) [inline nomic/gpt4all/gpt4all.py:28]: wait is the gpt4all model the first model we trained? should we update this to a newer one?

**Review by zanussbaum — COMMENTED** [review]
Had a few comments, mostly on documentation + clarifications

File: nomic/atlas.py
Comment (AndriyMulyar) [reply nomic/atlas.py:81]: yeah, they just shouldn't be using map_embeddings or map_text.
its kinda a hard problem to solve.

File: nomic/data_operations.py
Comment (AndriyMulyar) [reply nomic/data_operations.py:602]: What should this be. The issue with tags is that we don't get global access of all tag state

File: nomic/data_operations.py
Comment (AndriyMulyar) [reply nomic/data_operations.py:433]: all of AtlasMapEmbeddings is not fully complete yet. it is blocked by ben implementing logic on the atlas cloud backend for writing embeddings more efficiently for export.

Once we have an efficient method pull out ambient embeddings, accessing with .ambient will be possible.

For now the existing logic where we download all embeddings to a directory is here.

File: nomic/gpt4all/gpt4all.py
Comment (AndriyMulyar) [reply nomic/gpt4all/gpt4all.py:28]: We should remove gpt4all from the repository, but lets do it in a separate PR

File: nomic/atlas.py
Comment (AndriyMulyar) [reply nomic/atlas.py:81]: I will add information that these are set in upload order.

File: nomic/data_operations.py
Comment (AndriyMulyar) [reply nomic/data_operations.py:602]: I will add a .df method for consistency

File: docs/data_exploration_cleaning_tagging_in_atlas.ipynb
Comment (bmschmidt) [inline docs/data_exploration_cleaning_tagging_in_atlas.ipynb:175]: Ideally `map.tags` would do the retrieval on its own, not requiring a call to `map.tags.get_tags()`.

File: docs/map_state/tags.md
Comment (bmschmidt) [reply docs/map_state/tags.md:16]: We should make sure all our examples show people uploading ids.

With the new auto assignment it's harder to screw it up... but still not great.

File: docs/map_state/topics.md
Comment (bmschmidt) [inline docs/map_state/topics.md:1]: 'regions of homogenous color' -> 'regions of the same color'

File: docs/release_notes.md
Comment (bmschmidt) [inline docs/release_notes.md:20]: Better--"The `Accessing Atlas State` section of the documentation shows some of the ways you can use this data."

File: nomic/atlas.py
Comment (bmschmidt) [reply nomic/atlas.py:81]: I think we could write a warning to stderr on map_embeddings and map_text.

File: nomic/gpt4all/gpt4all.py
Comment (bmschmidt) [reply nomic/gpt4all/gpt4all.py:28]: Removal should definitely be part of the 2.0 release.

File: nomic/project.py
Comment (bmschmidt) [inline nomic/project.py:31]: We can lose this try-catch; pandas is a dependency again.

File: nomic/project.py
Comment (bmschmidt) [inline nomic/project.py:373]: Instead instantiate an AtlasProject and use the project.indices or get_map method to retrieve an AtlasProjection.
should be
    Instead instantiate an AtlasProject and use the project.maps attribute to retrieve an AtlasProjection.

File: nomic/project.py
Comment (bmschmidt) [inline nomic/project.py:1011]: Seeing how aggressively we use the word 'Index' here, I'm wondering if we should make atlas.maps return all the indices, rather than the projections.... I think it will be confusing for people to encounter the idea of an index if we're calling them 'maps'

File: docs/map_state/topics.md
Comment (AndriyMulyar) [reply docs/map_state/topics.md:15]: Resolved by adding examples containing the dataframe structure

File: nomic/atlas.py
Comment (AndriyMulyar) [reply nomic/atlas.py:81]: I will add a warning and also documentation.

File: nomic/gpt4all/gpt4all.py
Comment (AndriyMulyar) [reply nomic/gpt4all/gpt4all.py:28]: 🪓 done

File: nomic/data_operations.py
Comment (AndriyMulyar) [reply nomic/data_operations.py:602]: I have added a .df method which contains as columns a multi-hot vector where ones exist when the associated row has that tag.

File: docs/data_exploration_cleaning_tagging_in_atlas.ipynb
Comment (AndriyMulyar) [reply docs/data_exploration_cleaning_tagging_in_atlas.ipynb:175]: Done

File: docs/map_state/duplicates.md
Comment (AndriyMulyar) [reply docs/map_state/duplicates.md:7]: I think this would make a lot of sense.

File: docs/map_state/embeddings.md
Comment (AndriyMulyar) [reply docs/map_state/embeddings.md:4]: embedded sounds confusing (map.embeddings.embedded?)

i am fine with either latent/ambient.


Another idea: what if we called them
two_dimensional
high_dimensional

instead?

File: nomic/atlas.py
Comment (zanussbaum) [reply nomic/atlas.py:81]: What should people be using instead of map_embeddings or map_text then?

File: nomic/project.py
Comment (AndriyMulyar) [reply nomic/project.py:1011]: It is certainly confusing to have many abstraction, we have talked in person about this in-depth before.

Currently AtlasIndex is a proxy container from the ages where we were not certain what Atlas would be so built in room for flexibility.

I would propose we just *ignore* it and only mention AtlasProject and AtlasProjection ("map") across all documentation. We could essentially replace the word index to map in every location and no-one would be the wiser (and the code may be the clearer) as AtlasIndex is currently a passthrough object.

File: nomic/data_operations.py
Comment (zanussbaum) [reply nomic/data_operations.py:602]: yeah good point, resolving

**Comment by AndriyMulyar** [comment]
- Removed references to GPT4All
- Added warning when people don't specify ids
- Gave `map.tags` a dataframe that can be sliced and diced (but it is materialized from the server state on every call)
- Added more code/output examples for all data state classes

File: nomic/data_operations.py
Comment (zanussbaum) [inline nomic/data_operations.py:27]: i know these were probably a pain to write up but this is very much appreciated and really improves on the docs 😄

File: docs/map_state/embeddings.md
Comment (zanussbaum) [inline docs/map_state/embeddings.md:10]: Probably not the place for this, but most of my workflow has been creating a single map per project. What's the use case for multiple maps per single project?

File: nomic/atlas.py
Comment (zanussbaum) [inline nomic/atlas.py:91]: Should we mention the ID is in order from `range(len(data))` in order that the data was added?

**Review by zanussbaum — APPROVED** [review]
LGTM pending Ben's review

File: nomic/atlas.py
Comment (AndriyMulyar) [reply nomic/atlas.py:82]: p = AtlasProject()

p.add_text()

p.create_index()

File: nomic/data_operations.py
Comment (AndriyMulyar) [reply nomic/data_operations.py:27]: No, this is what the world needed

File: docs/map_state/embeddings.md
Comment (AndriyMulyar) [reply docs/map_state/embeddings.md:10]: if you index the project multiple times (say on different fields), you will have a map per index so possibly multiple indices

File: nomic/atlas.py
Comment (AndriyMulyar) [reply nomic/atlas.py:91]: updated

File: docs/map_state/embeddings.md
Comment (bmschmidt) [reply docs/map_state/embeddings.md:4]: I'm lukewarm on `two_dimensional`/`high_dimensional`. Partly because I don't like writing out numbers.

My thought on 'embedded' is that accessing embeddings does just by default work on the latent space, and the two-d space is a special case. It's not like 'embeddings.vector_search' lets you search in the 2d space as an option.

So maybe like `map.embeddings.array`, `map.embeddings.ndarray`, `map.embeddings.matrix` just returns the latent space as an `n x d` np.ndarray, and `map.embeddings.projected` returns the `n x 2` array.

File: nomic/project.py
Comment (bmschmidt) [reply nomic/project.py:1006]: That sounds good to me--so this method will be renamed `reuse_embeddings_from_map`?

File: docs/data_exploration_cleaning_tagging_in_atlas.ipynb
Comment (rguo123) [reply docs/data_exploration_cleaning_tagging_in_atlas.ipynb:175]: Notebook needs to be updated as well?

File: docs/map_state/duplicates.md
Comment (rguo123) [inline docs/map_state/duplicates.md:12]: super nitpick: In one of the tutorials, it may be good to showcase `get_map` instead of accessing the list of maps?

File: docs/map_state/topics.md
Comment (rguo123) [inline docs/map_state/topics.md:2]: Nit: "AtlasMap" since that's used elsewhere

File: docs/vector_database.md
Comment (rguo123) [inline docs/vector_database.md:12]: Nit: no period

File: nomic/pl_callbacks/pl_callback.py
Comment (rguo123) [inline nomic/pl_callbacks/pl_callback.py:59]: As part of docs, can we include what -1 is referring to here?

Comment (AndriyMulyar) [reply nomic/pl_callbacks/pl_callback.py:59]: done

**Pull request merged**
